### PR TITLE
Day off admin

### PR DESCRIPTION
**Pull request by nhi12345**

### CODE REVIEW

File: src/app/core/services/dayoff.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { TestBed } from '@angular/core/testing';
+
+import { DayoffService } from './dayoff.service';
+
+describe('DayoffService', () => {
+  beforeEach(() => TestBed.configureTestingModule({}));
+
+  it('should be created', () => {
+    const service: DayoffService = TestBed.get(DayoffService);
```
Comment (tranmanhlinh) [inline src/app/core/services/dayoff.service.spec.ts]: Please check and remove all test files if it is not neccessary.

File: src/app/core/services/dayoff.service.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  httpOptions = {
+    headers: new HttpHeaders({
+      'Context-Type': 'application/json',
+      'Authorization': 'Bearer '
+    })
+  };
+
+  constructor(
+    private http: HttpClient,
+    private alertService:AlertService
```
Comment (tranmanhlinh) [inline src/app/core/services/dayoff.service.ts]: Did you not use an auto format plugin for code editor?

File: src/app/core/services/dayofftype.service.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import { catchError } from 'rxjs/operators';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DayofftypeService {
+
+  private dayOffTypeUrl="https://helpdesk-kunlez-novahub.herokuapp.com/api/day_off_types"
+
```
Comment (tranmanhlinh) [inline src/app/core/services/dayofftype.service.ts]: Please store the url in file environment.ts for LOCAL and environment.prod.ts for production.

File: src/app/pages/admin-request/admin-request.component.ts [hunk 4]
```diff
@@ -65,9 +65,9 @@ export class AdminRequestComponent implements OnInit {
   getStatus() {
     this.statusService.getStatusList().subscribe(listStatus => {
       this.statusList = listStatus;
-      this.status = this.statusList.map(o => {
-        return {label: o.name, value: o}
-      })
+      // this.status = this.statusList.map(o => {
+      //   return {label: o.name, value: o}
+      // })
```
Comment (tranmanhlinh) [inline src/app/pages/admin-request/admin-request.component.ts]: Remove comment.

File: src/app/pages/dayoff/dayoff.component.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  idUser:number;
+
+  constructor(
+    private dayOffService : DayoffService ,
+    private userService :UserService
+  ) { }
+
+  ngOnInit() {
+    this.getUsers();
+    // this.getDayOffByUser(this.idUser);
```
Comment (tranmanhlinh) [inline src/app/pages/dayoff/dayoff.component.ts]: remove comment

File: src/app/pages/dayoff/dayoff.component.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    this.isVisibleDayOff1=false;
+    this.dayOffService.getDayOff(id).subscribe(data => this.dayoff = data);
+    this.dayOffService.rejectDayOff(this.dayoff).subscribe(dayoff=>this.dayoff=dayoff); 
+  }
+  handleCancelDayOff1(): void {
+    this.isVisibleDayOff1 = false;
+  }
+  checkStatus(status:string){
+    if(status == 'PENDING'){
+      return true;
```
Comment (tranmanhlinh) [inline src/app/pages/dayoff/dayoff.component.ts]: Could you defined status as an enum?

File: src/app/pages/dayoff/dayoff.component.ts [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    }else{
+      return false;
+    }
+  }
+
+  checkData(data: DayOff[]){
+    return (data==null || data.length==0)
+  }
```
Comment (tranmanhlinh) [inline src/app/pages/dayoff/dayoff.component.ts]: Should be  "return !data || !data.length"

File: src/app/pages/dayofftype/dayofftype.component.ts [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        value.name = dayOffType.name;
+        this.dayOffTypeService.updateDayOffType(dayOffType).subscribe(dayofftype => this.dayOffType = dayofftype);
+      }
+      return true;
+    })
+  }
+
+  handleCancel1(): void {
+    this.isVisible1 = false;
+  }
```
Comment (tranmanhlinh) [inline src/app/pages/dayofftype/dayofftype.component.ts]: "handleCancel1"? Could you rename it to something more meaningful?

File: src/app/shared/models/date-off.ts [hunk 9]
```diff
@@ -1,3 +1,15 @@
...
+
+export interface DayOff{
+    id: number;
+    createAt: Date;
+    dayStartOff: Date;
+    dayEndOff: Date;
+    description: string;
+    userEntity: User;
+    status: Status;
+}
```
Comment (tranmanhlinh) [inline src/app/shared/models/date-off.ts]: Why you use a same name in the same file for them?

File: src/app/core/services/dayoff.service.spec.ts (hunk 1 above)
Comment (nhi12345) [reply src/app/core/services/dayoff.service.spec.ts]: done

File: src/app/core/services/dayoff.service.ts (hunk 2 above)
Comment (nhi12345) [reply src/app/core/services/dayoff.service.ts]: done

File: src/app/core/services/dayofftype.service.ts (hunk 3 above)
Comment (nhi12345) [reply src/app/core/services/dayofftype.service.ts]: done

File: src/app/pages/admin-request/admin-request.component.ts (hunk 4 above)
Comment (nhi12345) [reply src/app/pages/admin-request/admin-request.component.ts]: done

File: src/app/pages/dayoff/dayoff.component.ts (hunk 5 above)
Comment (nhi12345) [reply src/app/pages/dayoff/dayoff.component.ts]: done

File: src/app/pages/dayoff/dayoff.component.ts (hunk 6 above)
Comment (nhi12345) [reply src/app/pages/dayoff/dayoff.component.ts]: done

File: src/app/pages/dayoff/dayoff.component.ts (hunk 7 above)
Comment (nhi12345) [reply src/app/pages/dayoff/dayoff.component.ts]: done

File: src/app/pages/dayofftype/dayofftype.component.ts (hunk 8 above)
Comment (nhi12345) [reply src/app/pages/dayofftype/dayofftype.component.ts]: done

File: src/app/shared/models/date-off.ts (hunk 9 above)
Comment (nhi12345) [reply src/app/shared/models/date-off.ts]: done

**Pull request merged**